rasa/nlu/regex_message_handler.py

```diff
@@ -0,0 +1,292 @@
+        entities_str = match.group(ENTITIES)
+        if entities_str is None:
+            return []
+
+        try:
+            parsed_entities = json.loads(entities_str)
+            if not isinstance(parsed_entities, dict):
+                raise ValueError(
+                    f"Parsed value isn't a json object "
+                    f"(instead parser found '{type(parsed_entities)}')"
+                )
+        except (JSONDecodeError, ValueError) as e:
+            rasa.shared.utils.io.raise_warning(
+                f"Failed to parse arguments in line '{match.string}'. "
+                f"Failed to decode parameters as a json object (dict). "
+                f"Make sure the intent is followed by a proper json object (dict). "
+                f"Continuing without entities. "
+                f"Error: {e}",
+                docs=DOCS_URL_STORIES,
+            )
+            parsed_entities = dict()
+
+        # validate the given entity types
+        entity_types = set(parsed_entities.keys())
+        unknown_entity_types = entity_types.difference(domain.entities)
+        if unknown_entity_types:
+            rasa.shared.utils.io.raise_warning(
+                f"Failed to parse arguments in line '{match.string}'. "
+                f"Expected entities from {domain.entities} "
+                f"but found {unknown_entity_types}. "
+                f"Continuing without unknown entity types. ",
+                docs=DOCS_URL_STORIES,
+            )
+            parsed_entities = {
+                key: value
+                for key, value in parsed_entities
+                if key not in unknown_entity_types
+            }
+
+        # convert them into the list of dictionaries that we expect
+        entities: List[Dict[Text, Any]] = []
+        for entity_type, entity_values in parsed_entities.items():
+            if not isinstance(entity_values, list):
+                entity_values = [entity_values]
+            for entity_value in entity_values:
+                entities.append(
+                    {
+                        ENTITY_ATTRIBUTE_TYPE: entity_type,
+                        # TODO: in the original implementation these are start and
+                        # end indices of the *substring that matches the entity regex*
+                        # ... but it should be indices of tokens to which these
+                        # entities belong. There are no such tokens. Hence, I think we
+                        # should just skip those indices. But can we, or will something
+                        # complain? If something expects indices (despite not being
+                        # able to use them because there's no text), then we could
+                        # instead use more explicit "unknown" indicators (e.g. -1)
+                        # ENTITY_ATTRIBUTE_START: sidx,
+                        # ENTITY_ATTRIBUTE_END: eidx,  # can't be more specific
+                        ENTITY_ATTRIBUTE_VALUE: entity_value,
```

Comment:
IMO it's fine to just skip them in this case. That's one of the reasons for this architectural change — to get rid of redundant/useless stuff.

Reply:
@dakshvar22 -- FYI, we had talked about this 'classifier' before you want on holiday -- we'll remove the TEXT and remove the START/END since no downstream model would do anything meaningful with it (because if it does something with those indices, it would also expect tokens and tokens/text, but those aren't there)